metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -161,7 +161,6 @@ def _build_query_output_node(
                 len(metric_spec.filter_spec_set.all_filter_specs) > 0
                 or metric_spec.offset_to_grain is not None
                 or metric_spec.offset_window is not None
-                or metric_spec.alias is not None
```

Comment:
We do need to keep this line in - adding some other tests would demonstrate why. The SQL rendering tests are most important here. Can you add a few more tests?
1. A SQL rendering test for the simple case (similar to the integration test you added)
2. A SQL rendering test with an order by and where filter on the aliased metric (using the original metric name)
3. An integration test for the same scenario as 2
4. A SQL rendering test for a derived metric query with a metric alias
5. An integration test for the same scenario as 4

I think the new integration tests added above will fail because we'll render the alias in the subqueries, not just the outer query. So we need to remove the alias before passing it any of the inner nodes.